tests/scripts/analyze_outcomes.py

```diff
@@ -660,31 +660,28 @@ def main():
         options = parser.parse_args()
 
         if options.list:
-            for task in TASKS:
+            for task in KNOWN_TASKS:
                 Results.log(task)
             sys.exit(0)
 
-        result = True
-
-        if options.task == 'all':
-            tasks = TASKS.keys()
+        if options.specified_tasks == 'all':
+            tasks_list = KNOWN_TASKS.keys()
         else:
-            tasks = re.split(r'[, ]+', options.task)
+            tasks_list = re.split(r'[, ]+', options.specified_tasks)
+
+            for task in tasks_list:
+                if task not in KNOWN_TASKS:
```

Comment:
I thought I followed this requirement from #8301:
> There are no call to sys.exit except for fatal errors that prevent from running any kind of outcome analysis whatsoever

but since you are asking I suspect I didn't understand it properly. Should I terminate anyway the program here?

Reply:
Sorry I got confused here by the `that prevent from running any kind of outcome analysis whatsoever` part. I'm going to fix it as you suggested